go/types/value_store.go

```diff
@@ -353,8 +353,8 @@ func (lvs *ValueStore) checkChunksInCache(v Value, readValues bool) Hints {
 		}
 
 		targetType := getTargetType(reachable)
-		if !entry.Type().Equals(targetType) {
-			d.Panic("Value to write contains ref %s, which points to a value of a different type: %+v != %+v", reachable.TargetHash(), entry.Type(), targetType)
+		if entry.Type().Kind() != ValueKind && targetType.Kind() != ValueKind && !entry.Type().Equals(targetType) {
```

Comment:
I'm not sure that we ever actually did that. There was a comment in the JS saying we should, but looking at the code I just changed, I don't think that was actually happening.